compiler/noirc_frontend/src/elaborator/scope.rs

```diff
@@ -26,16 +23,6 @@ type Scope = GenericScope<String, ResolverMeta>;
 type ScopeTree = GenericScopeTree<String, ResolverMeta>;
 
 impl<'context> Elaborator<'context> {
-    pub(super) fn lookup<T: TryFromModuleDefId>(&mut self, path: Path) -> Result<T, ResolverError> {
-        let span = path.span();
-        let id = self.resolve_path_or_error(path)?;
-        T::try_from(id).ok_or_else(|| ResolverError::Expected {
-            expected: T::description(),
-            got: id.as_str().to_owned(),
-            span,
-        })
-    }
```

Comment:
Given that the lookup now returns `PathResolutionItem` instead of `ModuleDefId` I had two options:
1. Change `TryFromModuleDefId` to `TryFromPathResolutionItem`
2. Remove this method and inline it in the caller

I found that `lookup` was used 3 times, and with the recent "lookup type" change I think it went down to two... and inlining this and removing `TryFromModuleDefId` results in more or less the same amount of code, or maybe less.

But let me know if this isn't right and I'll bring it back.